CONTRIBUTING.md

```diff
@@ -205,4 +205,86 @@ protected $containers = [
 Finally, you can run the tests using
 ```sh
 docker run --rm -v $(pwd):$(pwd):rw -w $(pwd) -v /var/run/docker.sock:/var/run/docker.sock  php:7.4-cli-alpine sh -c "apk add docker-cli && vendor/bin/phpunit tests/SDKTest.php"
-```
\ No newline at end of file
+```
+
+## SDK Generator Interface
+
+* **spec** -- This object is derived from the appwrite swagger spec
+  * **title** -> The title of the SDK you are generating (normally used as package name.)
+  * **description** -> Description of Appwrite SDK
+  * **namespace** -> SDK Namespace
+  * **version** -> SDK Version
+  * **endpoint** -> Default Endpoint (example: "https://appwrite.io/v1")
+  * **host** -> Default Host (example: "appwrite.io")
+  * **basePath** -> Default Path to API (example: "/v1")
+  * **licenseName** -> Name of licence for SDK
+  * **licenseURL** -> URL to SDK Licence
+  * **contactName** -> Name of Person/Team that created the SDK
+  * **contactURL** -> URL to contact for help with the SDK
+  * **contactEmail** -> Email Address to Contact for help with the SDK
+  * **services** -> Array of Services. Each service contains the following:
+    *  **name** -> The name of the service
+    *  **methods** -> Array of Methods that can be used with the service
+       * **method**  ->  HTTP Method to call
+       * **path** -> Path to API without a basePath
+       * **fullPath** -> Path to API with basePath
+       * **name** -> Name of API Method
+       * **packaging** -> A flag to indicate if the files at a path need to be packaged as a tarfile  
+       * **title** -> Title of API Method
+       * **description** -> Description of API Method
+       * **security** -> 
```

Comment:
Security tells if the method requires authentication if im not mistaken. 
@eldadfux need your comments here

Reply:
A list of the required authentication methods. We usually use this array to generate the code examples. For most client sdks this will only include the Appwrite project, for server side it will usually include `project` + `key` or `project` + `jwt`